server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -64,12 +61,11 @@
     private final DateMathExpressionResolver dateMathExpressionResolver;
 
     public IndexNameExpressionResolver(Settings settings) {
-        dateMathExpressionResolver = new DateMathExpressionResolver(settings);
         expressionResolvers = Arrays.asList(
-                dateMathExpressionResolver,
-                new WildcardExpressionResolver());
+            dateMathExpressionResolver = new DateMathExpressionResolver(settings),
```

Comment:
++ agree. It was moved because I just cherrypicked the changes from https://github.com/elastic/elasticsearch/pull/34507/files and there was no conflict on this one
will fix